Topics/Tech_Stacks/PrismaORM.md

```diff
@@ -0,0 +1,148 @@
+
+You should run
+```
+npx prisma migrate dev --name init
+```
+To create your initial migration, when you're developing your backend you might need to make many changes to your database, and for that you should use `npx prisma db push`, but to do that you'll have to set up what's called a shadow database and that's a whole new can of worms so we won't go into that just yet. 
+
+Behind the scenes, the migrate command also ran `npx prisma generate`, which builds the JavaScript package you'll use to interact with your database and access nice features like autocompletion and type safety.
+
+## Using Prisma
+
+Now that you've done all the hardwork of setting up Prisma, here's your chance to actually use it. Let me just emphasize right here that Prisma is a backend only library. You can't run it client side, if you're calling calling the Prisma client in any code that will be served frontend it just won't work! 
+
+In any case, for this quickstart let's assume you're just writing a node script. Let's create a script file with this being the content:
+```js
+import { PrismaClient } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+async function main() {
+    ...
+}
+
+main()
+    .then(async () => {
+        await prisma.$disconnect()
+    })
+    .catch(async (e) => {
+        console.error(e)
+        await prisma.$disconnect()
+        process.exit(1)
+    })
+```
+What you need to know is that the `prisma` variable is where you're going to call all the prisma functions, and the bit at the end is just some boiler plate to ensure that you're not inhaling all of your database's available connections. 
+
+The way the prisma client works is that all the query logic and data is expressed through a (heavily) nested JavaScript object. 
+
+here is an example of how to create a new `User` record. 
+
+```js
+...
+async function main() {
+    const user = await prisma.user.create({
+        data: {
+            name: 'Matthew',
+            email: 'Matthew@utoronto.ca',
+        }
+    })
+}
+...
+```
+Getting all `User` records that has an email address with the domain 'utoronto.ca' and the name Matthew with their `Post`s included is the following:
+```js
+...
+async function main() {
+    const users = await prisma.user.findMany({
+        where: {
+            email: {
+                endsWith: 'utoronto.ca',
+            },
+            name: 'Matthew',
+        },
+        include: {
+            posts: true,
+        }
+    })
+}
+...
+```
+
+And so on, you get the gist. The most difficult part of Prisma is learning these functions. But in my experience they're pretty intuitive and you'll pick up on them much faster than SQL.
+
+This concludes this short quickstart, Prisma is a very useful library that's still under active development, I enjoyed using it for my projects and I encourage you to check it out as well. The [Prisma Documentation](https://www.prisma.io/docs) explains all of this better than I ever could. Good luck coding. 
```

Comment:
I think it would be really helpful if you included some links on how to extend your knowledge through more resources, maybe include some links to talk about testing? Or for the user to learn more about joins and other complex queries

Reply:
Sure, I guess it's pretty important for a user to actual know how relational database or database systems in general work to use an ORM